src/pages/patientView/clinicalInformation/PatientViewPageStore.ts

```diff
@@ -745,6 +749,59 @@ export class PatientViewPageStore {
         invoke: ()=>Promise.resolve(indexHotspotsData(this.hotspotData))
     });
 
+    readonly sampleToMutationnGenePanelData = remoteData<{[sampleId: string]: GenePanelData}>({
+        await:()=>[
+            this.mutationMolecularProfileId
+        ],
+        invoke: async() => {
+            if (this.mutationMolecularProfileId.result) {
+                return fetchGenePanelData(this.mutationMolecularProfileId.result, this.sampleIds);
+            }
+            return {};
+        }
+    }, {});
+
+    @computed get sampleToMutationGenePanelId() {
+        return _.mapValues(this.sampleToMutationnGenePanelData.result, (genePanelData) => genePanelData.genePanelId);
+    }
+
+    readonly sampleToDiscreteGenePanelData = remoteData<{[sampleId: string]: GenePanelData}>({
+        await:()=>[
+            this.molecularProfileIdDiscrete
+        ],
+        invoke: async() => {
+            if (this.molecularProfileIdDiscrete.result) {
+                return fetchGenePanelData(this.molecularProfileIdDiscrete.result, this.sampleIds);
+            }
+            return {};
+        }
+    }, {});
+
+    @computed get sampleToDiscreteGenePanelId() {
+        return _.mapValues(this.sampleToDiscreteGenePanelData.result, (genePanelData) => genePanelData.genePanelId);
+    }
+
+    readonly genePanelIdToPanel = remoteData<{[genePanelId: string]: GenePanel}>({
+        await:()=>[
+            this.sampleToMutationnGenePanelData,
+            this.sampleToDiscreteGenePanelData
+        ],
+        invoke: async() => {
+            const sampleGenePanelInfo = _.concat(_.values(this.sampleToMutationnGenePanelData.result), _.values(this.sampleToDiscreteGenePanelData.result));
+            const panelIds = _(sampleGenePanelInfo)
+                .map((genePanelData) => genePanelData.genePanelId)
+                .filter((genePanelId) => !noGenePanelUsed(genePanelId))
+                .value();
+            return fetchGenePanel(panelIds);
+        }
+    }, {});
+
+    @computed get genePanelIdToEntrezGeneIds() {
```

Comment:
I was about to comment that I think this should be a remoteData, but then it seems like this isn't used to maybe can delete it?

Reply:
@adamabeshouse 
Well, it will be used by the PRs of RFC45 that are based on this PR. It would have been better to include this line in those PRs, but it would for now be more efficient to leave it in. Is that ok with you?

Also it is not a `remoteData` since @computed from `this.genePanelIdToPanel` which is a remoteData. Or am I mistaken here?

Reply:
since genePanelIdToPanel is remotedata, this won't be ready until that's ready. so users might want to await this as well, and only show when it's ready, for example. what do you think?

Reply:
Ah I agree. I have updated the code to the remoteData version of it.

Reply:
the difference is that on a remoteData you can check the status, so your views can decide what to show in case the data is still loading. you can distinguish the cases of (1) empty data (2) still loading, whereas with @computed you cant

Reply:
Ah I see. Thnx for clearing that up.